pymc3/stats.py

```diff
@@ -59,44 +60,60 @@ def wrapped_f(pymc3_obj, *args, **kwargs):
 
 
 @statfunc
-def autocorr(x, lag=1):
-    """Sample autocorrelation at specified lag.
+def autocorr(x, lag=None):
+    """
+    Compute autocorrelation using FFT for every lag for the input array
+    https://en.wikipedia.org/wiki/Autocorrelation#Efficient_computation
 
     Parameters
     ----------
     x : Numpy array
         An array containing MCMC samples
-    lag : int
-        The desidered lag to take in consideration
+
+    Returns
+    -------
+    acorr: Numpy array same size as the input array
     """
-    S = autocov(x, lag)
-    return S[0, 1] / np.sqrt(np.prod(np.diag(S)))
+    y = x - x.mean()
+    n = len(y)
+    result = fftconvolve(y, y[::-1])
+    acorr = result[len(result) // 2:]
+    acorr /= (n - np.arange(n))
```

Comment:
super minor: `np.arange(n, 0, -1)` is about 2x as fast (~1µs vs 2µs for `n=1000`), but also modestly more readable.  (I guess `fftconvolve` takes the longest of this stuff?)